scarb/src/compiler/compilation_unit.rs

```diff
@@ -93,11 +89,11 @@ impl CompilationUnit {
         format!("{}-{}", self.main_package_id.name, self.digest())
     }
 
-    pub fn name(&self) -> String {
+    pub fn name(&self, is_sole_for_package: bool) -> String {
```

Comment:
I don't like the fact that this getter is getting an extra argument. Especially as we can define a clear logic, that should be computable within the old `is_sole_for_package` check: we can just not count test targets in that condition.